internal/builder/embedded/driver.go

```diff
@@ -108,7 +108,14 @@ func (d *driver) BuildAndPush(ctx context.Context, opts *config.BuildOptions) (*
 
 func (d *driver) build(ctx context.Context, image string, opts *config.BuildOptions) error {
 	// download and extract remote OCI context
-	extract, err := archive.FetchAndExtract(opts.ContextURL)
+	fetchCtx := ctx
+	if opts.ContextTimeout > 0 {
+		var cancel context.CancelFunc
+		fetchCtx, cancel = context.WithTimeout(ctx, opts.ContextTimeout)
+		defer cancel()
+	}
+
+	extract, err := archive.FetchAndExtract(fetchCtx, opts.ContextURL)
```

Comment:
Maybe this is stylistic, but I thought the value semantics would allow to just use `ctx`.

e.g.

```go
	if opts.ContextTimeout > 0 {
		var cancel context.CancelFunc
		ctx, cancel = context.WithTimeout(ctx, opts.ContextTimeout)
		defer cancel()
	}

	extract, err := archive.FetchAndExtract(ctx, opts.ContextURL)
```

Reply:
I didn't want it to be passed to other methods in this file (i.e. build), but I think that's an argument for passing the timeout through.